Apply simple suggestions (#1)
ARR4N/ACPs: ACPs/194-streaming-asynchronous-execution/README.md

## Abstract

> Streaming Asynchronous Execution (SAE) decouples consensus and execution by introducing a queue upon which consensus is performed.

Reviewer: I assume that removal of "transaction" was deliberate as the queue can be conceptualised w.r.t. either transactions or blocks?

Author: Yeah

1. Concurrent consensus and execution streams eliminate node context switching, reducing latency caused by each waiting on the other.
In particular, "VM time" (akin to CPU time) more closely aligns with wall time since it is no longer eroded by consensus.
This increases gas per wall-second even without an increase in gas per VM-second.
2. Lean, execution-only clients can rapidly execute the queue agreed upon by consensus, providing accelerated receipt issuance and state computation.
Without the need to compute state _roots_, such clients can eschew expensive Merkle data structures.
End users see expedited but identical transaction results.
3. Irregular stop-the-world events like database compaction are amortised over multiple blocks.

> 4. Introduces additional bursty throughput by eagerly accepting transactions, without a reduction in security guarantees.

Reviewer: "Additional" with respect to what?

```suggestion
4. Introduces bursty throughput by eagerly accepting transactions, without a reduction in security guarantees.
```

Author: ACP-176 already supports bursty throughput

Reviewer: Which will be replaced by this because $C$ / the leaky bucket won't be used. Not important enough to block this PR, but noting anyway.

Author: I think the reason I don't like "Introduces bursty throughput" is because it makes it sound like there wasn't previously bursty throughput.

Like, we could make this say that this design doesn't remove the already existing bursty throughput? But that doesn't really seam like a "Performance improvement".

Reviewer: Lets leave it as "additional" for now as it's otherwise too bike-shed-y.

> 5. Non-data-dependent transactions, such as EOA-to-EOA transfers of value, can be observed prior to execution.

Reviewer: I don't think "observed" is the correct word. How about:

```suggestion
5. Third-party accounting of non-data-dependent transactions, such as EOA-to-EOA transfers of value, can be performed prior to execution.
```

### Future features

Performing transaction execution after consensus sequencing allows the usage of consensus artifacts in execution. This unblocks some additional future improvements:

> 1. Exposing a realtime VRF during transaction execution.

Reviewer: ```suggestion
1. Exposing a real-time VRF during transaction execution.
```

> The finality guarantees of Snowman consensus remove any distinction between _safe_ and _finalized_. 
> Furthermore, the _latest_ block is not at risk of re-org, only of a negligible risk of data corruption local to the API node.

> ### Optimal block building

Reviewer: I thought we were going to move this to the appendix and change SHOULD to MAY. We should then also add the caveat of it possibly hindering dependent transactions.

Author: moved it to the appendix

This ACP modifies the meaning of multiple fields in the block. A comprehensive list of changes will be produced once a reference implementation is available.

Likely fields to change include:
- `stateRoot`
- `receiptsRoot`
- `logsBloom`

> - `gasUsed`

Reviewer: (fyi) I've thus far kept `gasLimit` unchanged (i.e. the txs being pushed to the queue), which adds an interesting possibility of `gasUsed > gasLimit` because they refer to different tranches of transactions.

1. Consumption of gas units (by definition of the gas limit); and
2. Gas excess $x$ (and therefore gas price) at the time of execution.

> For a queue of transactions $Q = \{i\}_{i \ge 0}$ the gas excess $x_j$ immediately prior to execution of transaction $j \in Q$ is a monotonic, non-decreasing function of the gas usage of all preceding transactions in the queue; i.e. $x_j~:=~f(\{g_i\}_{i<j})$.

Reviewer: In regular Markdown+LaTeX you'd be right, but GitHub strips backslashes so the first one is to escape the second one, which is to escape the brace 🤦

Author: huh
